plugins/themes/controllers/backgroundlayers_controller.py

```diff
@@ -56,115 +65,279 @@ def index(self):
-                backgroundlayer["requestEncoding"] = form.requestEncoding.data
-                backgroundlayer["style"] = form.style.data  
-                backgroundlayer["originX"] = float(form.originX.data)
-                backgroundlayer["originY"] = float(form.originY.data)
+            # TODO: tileMatrixPrefix ?
+            item["tileMatrixPrefix"] = ""
+            if form.tileMatrixSet.data:
+                item["tileMatrixSet"] = form.tileMatrixSet.data
+            if form.projection.data:
+                item["projection"] = form.projection.data
+            if form.format.data:
+                item["format"] = form.format.data
+            if form.requestEncoding.data:
+                item["requestEncoding"] = form.requestEncoding.data
+            if form.style.data:
+                item["style"] = form.style.data
+            if form.originX.data:
+                item["originX"] = float(form.originX.data)
+            if form.originY.data:
+                item["originY"] = float(form.originY.data)
+            if form.resolutions.data:
                 resolutions = [
                     float(x) for x in form.resolutions.data.split(",")
                 ]
-                backgroundlayer["resolutions"] = resolutions
+                item["resolutions"] = resolutions
+            if form.tileSize.data:
                 tilesize = [
                     int(x) for x in form.tileSize.data.split(",")
                 ]
-                backgroundlayer["tileSize"] = tilesize
-
-                if form.with_capabilities.data:
-                    backgroundlayer["capabilities"] = json.loads(
-                        form.capabilities.data)
+                item["tileSize"] = tilesize
+            if form.with_capabilities.data:
+                item["capabilities"] = json.loads(
+                    form.capabilities.data)
 
+        # edit background layer
+        if index:
+            action_name = "updated"
+            self.themesconfig["themes"]["backgroundLayers"][index] = item
+        # new background layer
+        else:
+            action_name = "created"
             self.themesconfig["themes"]["backgroundLayers"].append(
-                backgroundlayer)
-
-            if ThemeUtils.save_themesconfig(self.themesconfig, self.app, self.handler):
-                message = "Background layer '{0}' was created.\
-                        ".format(backgroundlayer["title"])
-                flash(message, "success")
-            else:
-                message = "Could not save background layer '{0}'.\
-                        ".format(backgroundlayer["title"])
-                flash(message, "error")
-
-            return redirect(url_for("backgroundlayers"))
+                item)
 
+        if ThemeUtils.save_themesconfig(self.themesconfig, self.app, self.handler):
+            message = "Background layer '{0}' {1}.\
+                    ".format(item["title"], action_name)
```

Comment:
This will result in an exception if `title` is missing when its form field is left blank.

Maybe use `item.get("title", ")` instead of `item["title"]`, or make the _Title_ form field mandatory.